File: ios/QuerosermB/Domain/Model/Exchange.swift
import Foundation

// MARK: - Domain Models (Pure Swift — sem dependência de UIKit/SwiftUI)

struct Exchange: Identifiable, Equatable, Hashable {
    let id: Int
    let name: String
    let logo: String
    let slug: String
    let description: String?
    let websiteURL: String?
    let makerFee: Double?
    let takerFee: Double?
    let dateLaunched: String?
    let spotVolumeUSD: Double?
}

/// `id` é o `crypto_id` da CMC; na lista de assets da exchange pode repetir (mesma moeda, carteiras distintas).
struct Currency: Identifiable, Equatable {
    let id: Int
    let name: String
    let symbol: String
    let priceUSD: Double?
    let balance: Double?

    init(id: Int, name: String, symbol: String, priceUSD: Double?, balance: Double?) {
        self.id = id
        self.name = name
        self.symbol = symbol
        self.priceUSD = priceUSD
        self.balance = balance
    }
}

File: android/app/src/main/java/br/com/querosermb/domain/model/Exchange.kt
package br.com.querosermb.domain.model

data class Exchange(
    val id: Int,
    val name: String,
    val logo: String,
    val slug: String,
    val description: String?,
    val websiteURL: String?,
    val makerFee: Double?,
    val takerFee: Double?,
    val dateLaunched: String?,
    val spotVolumeUSD: Double?
)
